sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java

```diff
@@ -195,11 +233,37 @@
      * the {@link Flux} containing {@link SecretBase base secret} as output. This operation can then be used to get
      * the full secret with its value from {@code secretBase}. </p>
      * <p><strong>Code Samples</strong></p>
-     * <pre>
-     * secretAsyncClient.listSecrets().subscribe(secretBase -&gt;
-     *     client.getSecret(secretBase).subscribe(secretResponse -&gt;
-     *       System.out.printf("Secret with name %s and value %s \n", secretResponse.value().name(), secretResponse.value().value())));
-     * </pre>
+     * {@codesnippet com.azure.keyvault.secrets.secretclient.getSecret#secretBase}
+     *
+     * @param secretBase The {@link SecretBase base secret} secret base holding attributes of the
+     * secret being requested.
+     * @return A {@link Mono} containing the requested {@link Secret secret}.
+     * @throws ResourceNotFoundException when a secret with {@link SecretBase#name() name} and {@link
+     * SecretBase#version() version} doesn't exist in the key vault.
+     * @throws HttpRequestException if {@link SecretBase#name()}  name} or {@link SecretBase#version()
+     * version} is empty string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Secret> getSecret(SecretBase secretBase) {
+        Objects.requireNonNull(secretBase, "The Secret Base parameter cannot be null.");
+        if (secretBase.version() == null) {
+            return getSecret(secretBase.name());
+        }
+        return getSecretWithResponse(secretBase.name(), secretBase.version())
+            .flatMap(FluxUtil::toMono);
```

Comment:
This whole method should just call `getSecretWithResponse(secretBase).flatMap(FluxUtil::toMono)`. Try to leave the duplication to a minimum.